backend/src/nodes/nodes/image/video_frame_iterator.py

```diff
@@ -210,12 +210,20 @@ async def run(self, path: str, context: IteratorContext) -> None:
         if video_stream is None:
             raise Exception("No video stream found in file")
 
-        width = int(video_stream["width"])
-        height = int(video_stream["height"])
-        fps = int(video_stream["r_frame_rate"].split("/")[0]) / int(
-            video_stream["r_frame_rate"].split("/")[1]
-        )
-        frame_count = int(video_stream["nb_frames"])
+        width = video_stream.get("width", None)
+        if width is None:
+            raise Exception("No width found in video stream")
+        width = int(width)
+        height = video_stream.get("height", None)
+        if height is None:
+            raise Exception("No height found in video stream")
+        height = int(height)
+        fps = video_stream.get("r_frame_rate", None)
+        if fps is None:
+            raise Exception("No fps found in video stream")
+        fps = int(fps.split("/")[0]) / int(fps.split("/")[1])
+        frame_count = video_stream.get("nb_frames", 999999999)
+        frame_count = int(frame_count)
```

Comment:
Is there some way to know the length of the video (e.g. in seconds)? If so, then we could calculate the number of frames by doing `length * fps`.

Reply:
I'll check, that's a good idea

Reply:
We do get a `duration`, i believe its in seconds